syntax = "proto3";

package nameko;

service notif {
  rpc SendMessageToUsers (MessageToUsersRequest) returns (MessageToUsersReply) {}
  rpc GetUsersByNationalId(NationalIdRequest) returns (UserList) {}
}

message User {
  string name = 1;
  string national_id = 2;
  string phone = 3;
  string message = 4;
}

message NationalIdRequest {
  string national_id = 1;
}

message UserList {
  repeated User users = 1;
}

message MessageToUsersRequest {
  string message = 1;
  repeated User users = 2;
}

message MessageToUsersReply {
  string response = 1;
}
